meson.build

```diff
@@ -0,0 +1,177 @@
+project('microDNS', ['c'],
+  version : '0.1.0',
+  meson_version : '>= 0.51.0',
+  default_options : ['warning_level=2',
+                     'buildtype=release',
+                     'b_ndebug=if-release'])
+
+mdns_version = meson.project_version()
+
+ver_arr = mdns_version.split('.')
+
+mdns_major_version = ver_arr[0]
+mdns_minor_version = ver_arr[1]
+mdns_micro_version = ver_arr[2]
+
+cc = meson.get_compiler('c')
+
+warning_flags = [
+    '-Wall',
+    '-Wsign-compare',
+    '-Wextra',
+    '-Wstrict-aliasing',
+    '-Wstrict-overflow',
+    '-Wformat=2',
+    '-Wno-unused-parameter',
+    '-pipe',
+    '-Wcast-align',
+    '-Wpointer-arith',
+    '-Wmissing-prototypes',
+    '-Wwrite-strings',
+    '-Wlogical-op',
+]
+
+add_project_arguments(cc.get_supported_arguments(warning_flags), language: 'c')
+
+cdata = configuration_data()
+
+deps = []
+
+host_system = host_machine.system()
+
+if host_system == 'windows'
+    deps += [cc.find_library('ws2_32')]
+    deps += [cc.find_library('iphlpapi')]
+endif
+
+inet_ntop_src = '''
+#ifdef _WIN32
+#include <ws2tcpip.h>
+#include <windows.h>
+# if _WIN32_WINNT < 0x600
+#  error Needs vista+
+# endif
+#else
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#endif
+int main() {}
+'''
+
+cdata.set10('HAVE_INET_NTOP', cc.links(inet_ntop_src, dependencies: deps))
+
+poll_src = '''
+#include <stddef.h>
+#ifdef _WIN32
+#include <winsock2.h>
+#include <windows.h>
+# if _WIN32_WINNT < 0x600
+#  error Needs vista+
+# endif
+#else
+#include <poll.h>
+#endif
+int main() {}
+'''
+
+cdata.set10('HAVE_INET_NTOP', cc.links(inet_ntop_src, dependencies: deps))
+cdata.set10('HAVE_POLL', cc.links(poll_src, dependencies: deps))
+
+configure_file(output : 'config.h', configuration : cdata)
+
+c_args = ['-DHAVE_CONFIG_H']
+
+libcompat_sources = [
+    'compat/compat.c',
+    'compat/inet.c',
+    'compat/poll.c',
+]
+
+incdirs = include_directories('src', 'compat')
+
+libcompat = static_library('compat', libcompat_sources,
+    include_directories: incdirs,
+    dependencies: deps,
+    c_args: c_args
+)
+
+libmicrodns_sources = [
+    'src/mdns.c',
+    'src/rr.c'
+]
+
+libmicrodns = library('microdns', libmicrodns_sources,
+    include_directories: incdirs,
+    link_with: libcompat,
+    dependencies: deps,
+    c_args: c_args,
+)
+
+examples = [
+    ['listen', ['examples/main.c']],
+    ['announce', ['examples/announce.c']],
+]
+
+foreach example: examples
+    executable(example[0], example[1],
+        include_directories: incdirs,
+        link_with: libmicrodns,
+    )
+endforeach
+
+pkg_cdata = configuration_data()
+
+pkg_cdata.set('prefix', join_paths(get_option('prefix')))
+pkg_cdata.set('exec_prefix', '${prefix}')
+pkg_cdata.set('libdir', '${prefix}/@0@'.format(get_option('libdir')))
+pkg_cdata.set('includedir', '${prefix}/@0@'.format(get_option('includedir')))
+pkg_cdata.set('VERSION', mdns_version)
+pkg_cdata.set('LIBSOCKET', host_system == 'windows' ? '-lws2_32 -liphlpapi': '')
+pkg_cdata.set('LIBS', '')
```

Comment:
What is this used for?

Reply:
we actually don't need that anymore, as this commit removes the autotools build system, so I can just modify the pc.in file without touching autotools

Reply:
Fixed